qa/tasks/cephfs/kernel_mount.py

```diff
@@ -13,32 +17,48 @@
 
 
 class KernelMount(CephFSMount):
-    def __init__(self, ctx, test_dir, client_id, client_remote, brxnet):
-        super(KernelMount, self).__init__(ctx, test_dir, client_id, client_remote, brxnet)
+    def __init__(self, ctx, test_dir, client_id, client_remote,
+                 client_keyring_path=None, hostfs_mntpt=None,
+                 cephfs_name=None, cephfs_mntpt=None, brxnet=None):
+        super(KernelMount, self).__init__(ctx=ctx, test_dir=test_dir,
+            client_id=client_id, client_remote=client_remote,
+            client_keyring_path=client_keyring_path, hostfs_mntpt=hostfs_mntpt,
+            cephfs_name=cephfs_name, cephfs_mntpt=cephfs_mntpt, brxnet=brxnet)
 
-    def mount(self, mount_path=None, mount_fs_name=None, mountpoint=None, mount_options=[]):
-        if mountpoint is not None:
-            self.mountpoint = mountpoint
-        self.setupfs(name=mount_fs_name)
-        self.setup_netns()
+        self.mounted = False
 
-        log.info('Mounting kclient client.{id} at {remote} {mnt}...'.format(
-            id=self.client_id, remote=self.client_remote, mnt=self.mountpoint))
+    def mount(self, mntopts=[], createfs=True, **kwargs):
+        self.update_attrs(**kwargs)
+        self.assert_and_log_minimum_mount_details()
 
-        self.client_remote.run(args=['mkdir', '-p', self.mountpoint],
-                               timeout=(5*60))
+        self.setup_netns()
+
+        # TODO: don't call setupfs() from within mount(), since it's
+        # absurd. The proper order should be: create FS first and then
+        # call mount().
+        if createfs:
+            self.setupfs(name=self.cephfs_name)
+        if not self.cephfs_mntpt:
+            self.cephfs_mntpt = '/'
 
-        if mount_path is None:
-            mount_path = "/"
+        stderr = StringIO()
+        try:
+            self.client_remote.run(args=['mkdir', '-p', self.hostfs_mntpt],
+                                   timeout=(5*60), stderr=stderr)
+        except CommandFailedError:
+            if 'file exists' not in stderr.getvalue().lower():
+                raise
 
-        opts = 'name={id},norequire_active_mds,conf={conf}'.format(id=self.client_id,
-                                                        conf=self.config_path)
+        opts = 'name=' + self.client_id
+        if self.client_keyring_path and self.client_id is not None:
+            opts += 'secret=' + self.get_key_from_keyfile()
```

Comment:
Are there any advantages to using `secretfile`? I feel using `secret` is better since the key gets logged which keeps things transparent and can help with debugging (otherwise someone might have to add a log message in future to print the content of the keyring path).

Reply:
Benefits of using `secretfile` for kernel mount
* It's  analagous to what's already done in `FuseMount` and `LocalFuseMount`
* tests what's actually recommended in the real world for security reasons
* no need to add a new `get_key_from_keyfile()` method